app/lib/publication_id_validator.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const request = require('request');
+
+/**
+ * DOI - Digital Object Identifier
+ *
+ * 10.1594/GFZ.GEOFON.gfz2009kciu
+ */
+const DOI_VALIDATOR = /^(10\.\d{4,9}\/.+)$/;
+
+/**
+ * Pubmed ID
+ *
+ * 123456789
+ */
+const PUBMED_VALIDATOR = /^(\d+)$/;
+
+const DOI_BASE_URL = 'http://doi.org/api/handles/';
+const PUBMED_BASE_URL = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&retmode=json&id=';
+
+function doiUrl(doi) {
+	return `${DOI_BASE_URL}${doi}`;
```

Comment:
https://nodejs.org/api/url.html#url_url_resolve_from_to

You might want to use `url.parse()` or `url.format()` here, built in to Node.